sway/input_state.c

```diff
@@ -48,27 +48,261 @@ void release_key(keycode key) {
+			changed = true;
+		}
+		if (initial.y + dy != initial.ptr->y) {
+			initial.ptr->y = initial.y + dy;
+			changed = true;
+		}
+		update_geometry(initial.ptr);
+		break;
+
+	case M_FLOATING | M_RESIZING:
+		dx -= pointer_state.right.x;
+		dy -= pointer_state.right.y;
+		initial.ptr = pointer_state.right.view;
+		if (lock.left) {
+			if (initial.w + dx > min_sane_w) {
+				initial.ptr->width = initial.w + dx;
+			}
+		} else { //lock.right
+			if (initial.w - dx > min_sane_w) {
+				initial.ptr->width = initial.w - dx;
+				initial.ptr->x = initial.x + dx;
+			}
+		}
+		if (lock.top) {
+			if (initial.h + dy > min_sane_h) {
+				initial.ptr->height = initial.h + dy;
+			}
+		} else { //lock.bottom
+			if (initial.h - dy > min_sane_h) {
+				initial.ptr->height = initial.h - dy;
+				initial.ptr->y = initial.y + dy;
+			}
+		}
+		update_geometry(initial.ptr);
+		break;
+
+	case M_TILING | M_DRAGGING:
+		// swap current view under pointer with dragged view
+		if (pointer_state.view && pointer_state.view != initial.ptr) {
+			// Swap them around
+			swap_container(pointer_state.view, initial.ptr);
+			update_geometry(pointer_state.view);
+			update_geometry(initial.ptr);
+			// Set focus back to initial view
+			set_focused_container(initial.ptr);
+		}
+		break;
+
+	case M_TILING | M_RESIZING:
+		dx -= pointer_state.right.x;
+		dy -= pointer_state.right.y;
+		// resize if we can
+		if (initial.lr.ptr) {
+			if (lock.left) {
+				// Check whether its fine to resize
+				if (initial.w + dx > min_sane_w && initial.lr.w - dx > min_sane_w) {
+					initial.ptr->width = initial.w + dx;
+					initial.lr.ptr->width = initial.lr.w - dx;
+					initial.lr.ptr->x = initial.lr.x + dx;
+				}
+			} else { //lock.right
+				if (initial.w - dx > min_sane_w && initial.lr.w + dx > min_sane_w) {
+					initial.ptr->width = initial.w - dx;
+					initial.ptr->x = initial.x + dx;
+					initial.lr.ptr->width = initial.lr.w + dx;
+				}
+				changed = true;
+			}
+			arrange_windows(initial.lr.ptr->parent, -1, -1);
+		}
+		if (initial.tb.ptr) {
+			if (lock.top) {
+				if (initial.h + dy > min_sane_h && initial.tb.h - dy > min_sane_h) {
+					initial.ptr->height = initial.h + dy;
+					initial.tb.ptr->height = initial.tb.h - dy;
+					initial.tb.ptr->y = initial.tb.y + dy;
+				}
+			} else { //lock.bottom
+				if (initial.h - dy > min_sane_h && initial.tb.h + dy > min_sane_h) {
+					initial.ptr->height = initial.h - dy;
+					initial.ptr->y = initial.y + dy;
+					initial.tb.ptr->height = initial.tb.h + dy;
+				}
+				changed = true;
+			}
+			arrange_windows(initial.tb.ptr->parent, -1, -1);
+		}
+		if (changed) {
+			arrange_windows(initial.ptr->parent, -1, -1);
+		}
+		changed = false;
+	default:
+		return;
+	}
+}
+
+void pointer_mode_reset(void) {
+	switch (pointer_state.mode) {
+	case M_FLOATING | M_DRAGGING:
+	case M_FLOATING | M_RESIZING:
+		reset_initial_view();
+		break;
+
+	case M_TILING | M_DRAGGING:
+	case M_TILING | M_RESIZING:
```

Comment:
might reset tiling resize the same way as floating, all the information we need is there. just like 5 loc